src/modules/Tickets/TicketController.ts

```diff
@@ -7,46 +7,68 @@ import {
   Post,
   Put,
   Delete,
+  JsonController,
 } from "routing-controllers";
 
 import { TicketServices } from "./TicketServices";
+import { Container } from "typedi";
+import { ITicket } from "@app/data/abstraction/entities/ITicket";
+import _ from "lodash";
+import { NotAllow } from "@app/decorator/notAllow";
 
 @Controller()
 export class TicketController {
+  serviceInstance = Container.get(TicketServices);
+
   @Get("/")
-  response() {
-    return { msg: "Welcome" };
+  async welcomeMessage() {
+    try {
+      return "Welcome";
+    } catch (error) {
+      return;
+    }
   }
+
   @Get("/tickets")
-  getAll() {
-    const services = new TicketServices();
-    return services.getAll();
+  async fetchAllTickets() {
+    try {
+      const services = await this.serviceInstance.fetchAllTickets();
+      return services;
+    } catch (error) {
+      return error;
+    }
   }
 
-  @Get("/ticket/:id")
-  getOne(@Param("id") id: string) {
-    const services = new TicketServices();
-    return services.findTicket(id);
+  @Get("/tickets/:id")
+  async findTicketById(@Param("id") id: string) {
+    const services = await this.serviceInstance.findTicketById(id);
+    return services;
   }
 
-  @Post("/ticket")
-  async post(@Body() user: any) {
-    const services = new TicketServices();
-    await services.createTicket(user);
-    return { msg: "successfully crates" };
+  @Post("/tickets")
+  async createTicket(@Body() user: ITicket) {
+    try {
+      await this.serviceInstance.createTicket(user);
+      return { msg: "successfully crates" };
+    } catch (error) {
+      return error;
```

Comment:
returning an Error object may resulting status 200 with an error body, please consider to rethrow the exception when it is `HttpError` or `new HttpError(500)` for the unknown/unexpected errors.